src/workers/cdc/primary.ts

```diff
@@ -1223,6 +1227,49 @@ const onContentPreferenceChange = async (
   }
 };
 
+const onOpportunityMatchChange = async (
+  con: DataSource,
+  logger: FastifyBaseLogger,
+  data: ChangeMessage<OpportunityMatch>,
+) => {
+  if (data.payload.op === 'u') {
+    if (
+      data.payload.after?.status === OpportunityMatchStatus.CandidateAccepted &&
+      data?.payload.before?.status !== OpportunityMatchStatus.CandidateAccepted
+    ) {
+      const candidatePreference = await con
+        .getRepository(UserCandidatePreference)
+        .findOneBy({
+          userId: data.payload.after.userId,
+        });
+
+      if (!candidatePreference) {
+        logger.warn(
+          { userId: data.payload.after.userId },
+          'Candidate preference not found for user accepting opportunity',
+        );
+        return;
+      }
+
+      await triggerTypedEvent(
+        logger,
+        'api.v1.candidate-accepted-opportunity',
+        new CandidateAcceptedOpportunityMessage({
+          opportunityId: data.payload.after.opportunityId,
+          userId: data.payload.after.userId,
+          createdAt: data.payload.after.createdAt,
+          updatedAt: data.payload.after.updatedAt,
+          screening: data.payload.after.screening,
+          candidatePreference: {
+            ...candidatePreference,
+            updatedAt: getSecondsTimestamp(candidatePreference.updatedAt),
+          },
+        }),
```

Comment:
maybe pull this out as reusable function?

Reply:
because then we could re-use it for declined opportunity just by changing the topic name